tests/e2e/vms/xsvm.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/ava-labs/avalanchego/tests/fixture/tmpnet"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/api"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/cmd/issue/export"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/cmd/issue/importtx"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/cmd/issue/transfer"
+	"github.com/ava-labs/avalanchego/vms/example/xsvm/genesis"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+)
+
+var (
+	subnetAName = "xsvm-a"
+	subnetBName = "xsvm-b"
+)
+
+func XSVMSubnets(nodes ...*tmpnet.Node) []*tmpnet.Subnet {
+	return []*tmpnet.Subnet{
+		newXSVMSubnet(subnetAName, nodes...),
+		newXSVMSubnet(subnetBName, nodes...),
+	}
+}
+
+var _ = ginkgo.Describe("[XSVM]", func() {
+	require := require.New(ginkgo.GinkgoT())
+
+	ginkgo.It("should support transfers between subnets", func() {
+		network := e2e.Env.GetNetwork()
+
+		sourceSubnet := network.GetSubnet(subnetAName)
+		require.NotNil(sourceSubnet)
+		destinationSubnet := network.GetSubnet(subnetBName)
+		require.NotNil(destinationSubnet)
+
+		sourceChain := sourceSubnet.Chains[0]
+		destinationChain := destinationSubnet.Chains[0]
+
+		apiNode := network.Nodes[0]
+		tests.Outf(" issuing transactions on %s (%s)\n", apiNode.NodeID, apiNode.URI)
+
+		destinationKey, err := secp256k1.NewPrivateKey()
+		require.NoError(err)
+
+		ginkgo.By("checking that the funded key has sufficient funds for the export")
+		sourceClient := api.NewClient(apiNode.URI, sourceChain.ChainID.String())
+		initialSourcedBalance, err := sourceClient.Balance(
+			e2e.DefaultContext(),
+			sourceChain.PreFundedKey.Address(),
+			sourceChain.ChainID,
+		)
+		require.NoError(err)
+		require.GreaterOrEqual(initialSourcedBalance, units.Schmeckle)
+
+		ginkgo.By(fmt.Sprintf("exporting from chain %s on subnet %s", sourceChain.ChainID, sourceSubnet.SubnetID))
+		exportTxStatus, err := export.Export(
+			e2e.DefaultContext(),
+			&export.Config{
+				URI:                apiNode.URI,
+				SourceChainID:      sourceChain.ChainID,
+				DestinationChainID: destinationChain.ChainID,
+				Amount:             units.Schmeckle,
+				To:                 destinationKey.Address(),
+				PrivateKey:         sourceChain.PreFundedKey,
+			},
+		)
+		require.NoError(err)
+		tests.Outf(" issued transaction with ID: %s\n", exportTxStatus.TxID)
+
+		ginkgo.By("checking that the export transaction has been accepted on all nodes")
+		for _, node := range network.Nodes[1:] {
+			client := api.NewClient(node.URI, sourceChain.ChainID.String())
+			e2e.Eventually(func() bool {
+				currentNonce, err := client.Nonce(e2e.DefaultContext(), destinationKey.Address())
+				require.NoError(err)
+				return currentNonce >= exportTxStatus.Nonce
+			}, e2e.DefaultTimeout, e2e.DefaultPollingInterval, "failed to see export transaction on all nodes before timeout")
```

Comment:
Should be be re-using `api.WaitForAcceptance`?

Reply:
We should be using `exportTxStatus.Nonce` and using `WaitForAcceptance` - no?

Reply:
I don't think there are different requirements. Shouldn't we be using `>` here?

Reply:
See my above comment - we aren't using the right key here (which is why `>=` is needed rather than `>`).